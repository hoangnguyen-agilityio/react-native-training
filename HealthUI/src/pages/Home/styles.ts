import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

const useStyle = () => {
  const styles = StyleSheet.create({
    root: {
      backgroundColor: colors.white,
      flex: 1,
    },
    flexRow: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'stretch',
      marginTop: 24,
    },
    flexItemLeft: {
      flex: 1,
      marginRight: 12,
    },
    flexItemRight: {
      flex: 1,
      marginLeft: 12,
    },
    itemsCenter: {
      alignItems: 'center',
    },
    waterIllusion: {
      marginTop: 30,
      marginBottom: 10,
      alignItems: 'center',
    },
    caloriesGraph: {
      width: '100%',
      height: 'auto',
      aspectRatio: 2.9,
      marginBottom: 85,
      marginTop: 45,
    },
    number: {
      fontFamily: 'SF-Pro-Display-Bold',
      fontSize: 16,
      lineHeight: 16,
      color: colors.purple,
    },
    measure: {
      fontFamily: 'SF-Pro-Display-Regular',
      fontSize: 12,
      lineHeight: 14,
      color: '#A274CD',
    },
  });

  return styles;
};

export default useStyle;
