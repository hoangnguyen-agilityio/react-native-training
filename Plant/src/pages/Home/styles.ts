import { StyleSheet } from 'react-native';
import COLORS from '../../constants/colors';

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  headerContent: {
    marginTop: 75,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  headerTitle: {
    fontFamily: 'SF-Pro-Display-Bold',
    fontSize: 21,
    lineHeight: 21,
    marginBottom: 10,
    color: COLORS.WHITE,
  },
  headerDesc: {
    fontFamily: 'SF-Pro-Display-Regular',
    fontSize: 14,
    lineHeight: 14,
    color: COLORS.WHITE,
  },
  avatar: {
    width: 47,
    height: 47,
  },
  main: {
    backgroundColor: COLORS.WHITE,
  },
  catalogsWrapper: {
    marginTop: 35,
    marginHorizontal: -6,
  },
  sectionTitle: {
    marginTop: 24,
    marginBottom: 16,
    fontFamily: 'SF-Pro-Display-Bold',
    fontSize: 17,
    lineHeight: 20,
    color: '#36455A'
  },
  horizontalListWrapper: {
    marginRight: -24,
  }
});

export default styles;
