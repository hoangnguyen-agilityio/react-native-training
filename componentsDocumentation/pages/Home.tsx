import React from 'react';
import { SafeAreaView, StyleSheet, FlatList, Button, View } from 'react-native';
import { ComponentListType, COMPONENT_LIST } from '../configs/componentList';
import type { NativeStackNavigationProp } from '@react-navigation/native-stack';

interface Item {
  item: {
    page: keyof ComponentListType;
    title: string;
    id: string;
  };
}

interface IPdpPageProps {
  navigation: NativeStackNavigationProp<ComponentListType, 'Home'>;
}

const HomePage = ({ navigation }: IPdpPageProps) => {
  const Item = ({ item }: Item) => (
    <View style={styles.listItem}>
      <Button
        onPress={() => navigation.navigate(item.page, { name: item.title })}
        title={item.title}
      />
    </View>
  );

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={COMPONENT_LIST}
        renderItem={Item}
        keyExtractor={item => item.id}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  listItem: {
    fontSize: 24,
    fontWeight: '600',
    marginBottom: 16,
  },
});

export default HomePage;
