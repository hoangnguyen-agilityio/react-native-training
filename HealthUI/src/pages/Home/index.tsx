import React from 'react';
import { ScrollView, View, Image, Text } from 'react-native';
import { AnimatedCircularProgress } from 'react-native-circular-progress';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import FeatureLayout from '../../components/FeatureLayout';
import SectionLayout from '../../components/SectionLayout';
import UserBanner from '../../components/UserBanner';
import WaterIcon from '../../assets/water.svg';
import HeartIcon from '../../assets/duotone.svg';
import HeartGraph from '../../assets/heart-graph.svg';
import CaloriesIcon from '../../assets/calories.svg';
import WalkIcon from '../../assets/walk.svg';
import BedIcon from '../../assets/bed.svg';
import styles from './styles';
import TabBar from '../../components/TabBar';

import { NavigationListType } from '../../constants/navigationList';

interface Props {
  navigation: NativeStackNavigationProp<NavigationListType, 'Home'>;
}

const Home = ({ navigation }: Props) => {
  return (
    <View style={styles.root}>
      <ScrollView contentContainerStyle={{ paddingTop: 24, paddingBottom: 24 }}>
        <SectionLayout>
          <UserBanner
            title="Hi, Nam Duong"
            date="Wednesday 29 Dec"
            avatarUrl={require('../../assets/user.png')}
          />
        </SectionLayout>
        <SectionLayout>
          <View style={styles.flexRow}>
            <FeatureLayout
              title="Heart"
              icon={<HeartIcon />}
              number="105"
              measure="bmp"
              theme="dark"
              customStyle={styles.flexItemLeft}>
              <HeartGraph width={'100%'} />
            </FeatureLayout>
            <FeatureLayout
              title="Water"
              icon={<WaterIcon />}
              number="1.0"
              measure="liters"
              customStyle={styles.flexItemRight}
              onPress={() => navigation.navigate('Onboarding')}>
              <View style={styles.waterIllusion}>
                <Image source={require('../../assets/water-illu.png')} />
              </View>
            </FeatureLayout>
          </View>

          <View style={styles.flexRow}>
            <View style={styles.flexItemLeft}>
              <FeatureLayout
                title="Walk"
                icon={<WalkIcon />}
                customStyle={{ flex: 2, marginBottom: 12, paddingBottom: 16 }}>
                <View style={{ alignItems: 'center' }}>
                  <AnimatedCircularProgress
                    size={90}
                    rotation={0}
                    width={6}
                    fill={60}
                    lineCap="round"
                    tintColor="#5142AB"
                    backgroundColor="#F3F4F7">
                    {fill => (
                      <View style={{ alignItems: 'center' }}>
                        <Text style={styles.number}>5460</Text>
                        <Text style={styles.measure}>steps</Text>
                      </View>
                    )}
                  </AnimatedCircularProgress>
                </View>
              </FeatureLayout>

              <FeatureLayout
                title="Sleep"
                icon={<BedIcon />}
                number="06:32"
                measure="hours"
                customStyle={{ flex: 1, marginTop: 12 }}
              />
            </View>
            <FeatureLayout
              title="Calories"
              icon={<CaloriesIcon />}
              number="540"
              measure="kcal"
              linearGradientColors={['#FFFFFF', '#EDEFF7']}
              onPress={() =>
                navigation.navigate('Calories', { title: 'Calories' })
              }
              customStyle={{
                ...styles.flexItemRight,
                backgroundColor:
                  'linear-gradient(360deg, #FFFFFF 0%, #EDEFF7 108.19%)',
              }}>
              <Image
                style={styles.caloriesGraph}
                source={require('../../assets/line.png')}
              />
            </FeatureLayout>
          </View>
        </SectionLayout>
      </ScrollView>
      <TabBar />
    </View>
  );
};

export default Home;
