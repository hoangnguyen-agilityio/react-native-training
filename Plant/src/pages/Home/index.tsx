import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import React, { FC, useEffect, useState } from 'react';
import {
  Image,
  Linking,
  Pressable,
  ScrollView,
  Text,
  View,
} from 'react-native';
import Catalogs from '../../components/Catalogs';
import Header from '../../components/HeaderLayout';
import Photographies from '../../components/Photographies';
import PlantTypes from '../../components/PlantTypes';
import SearchBar from '../../components/SearchBar';
import Section from '../../components/SectionLayout';
import { NavigationListType } from '../../constants/navigationList';
import styles from './styles';
import { Notification, Notifications } from 'react-native-notifications';

interface Props {
  navigation: NativeStackNavigationProp<NavigationListType, 'Home'>;
}

const useMount = func => useEffect(() => func(), []);

const useInitialURL = () => {
  const [url, setUrl] = useState(null);
  const [processing, setProcessing] = useState(true);

  useMount(() => {
    const getUrlAsync = async () => {
      // Get the deep link used to open the app
      const initialUrl = await Linking.getInitialURL();

      // The setTimeout is just for testing purpose
      setTimeout(() => {
        setUrl(initialUrl);
        setProcessing(false);
      }, 1000);
    };

    getUrlAsync();
  });

  return { url, processing };
};

const Home: FC<Props> = ({ navigation }) => {
  const { url: initialUrl, processing } = useInitialURL();

  useEffect(() => {
    Notifications.registerRemoteNotifications();

    Notifications.events().registerNotificationReceivedForeground(
      (notification: Notification, completion) => {
        console.log(
          `Notification received in foreground: ${notification.title} : ${notification.body}`,
        );
        completion({ alert: false, sound: false, badge: false });
      },
    );

    Notifications.events().registerNotificationOpened(
      (notification: Notification, completion) => {
        console.log(`Notification opened: ${notification.payload}`);
        completion();
      },
    );
  }, []);

  const handleMoveToPage = (page: keyof NavigationListType) => {
    if (page) {
      navigation.navigate(page);
    }
  };

  return (
    <ScrollView>
      <Header textBg="Home">
        <Section>
          <Text>
            {processing
              ? `Processing the initial url from a deep link`
              : `The deep link is: ${initialUrl || 'None'}`}
          </Text>
        </Section>
        <Section>
          <Pressable
            style={styles.headerContent}
            onPress={() =>
              Notifications.postLocalNotification({
                title: 'Local notification',
                body: 'This notification was generated by the app!',
                extra: 'data',
              })
            }>
            <View>
              <Text style={styles.headerTitle}>Hello Taylor,</Text>
              <Text style={styles.headerDesc}>
                Letâ€™s Learn More About Plants
              </Text>
            </View>
            <Image
              source={require('../../assets/images/avatar.png')}
              style={styles.avatar}
            />
          </Pressable>
        </Section>
      </Header>
      <Section style={styles.main}>
        <SearchBar style={{ marginTop: -25 }} />
        <View style={styles.catalogsWrapper}>
          <Catalogs handleMoveToPage={handleMoveToPage} />
        </View>

        <Text style={styles.sectionTitle}>Plant Types</Text>
        <View style={styles.horizontalListWrapper}>
          <PlantTypes />
        </View>
        <Text style={styles.sectionTitle}>Photography</Text>
        <View style={styles.horizontalListWrapper}>
          <Photographies />
        </View>
      </Section>
    </ScrollView>
  );
};

export default Home;
