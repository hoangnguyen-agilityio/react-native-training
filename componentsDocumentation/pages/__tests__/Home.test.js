import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';
import {FlatList, Button} from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Home page will has a list with 9 items', () => {
  const tree = renderer.create(<Home />);
  const list = tree.root.findAllByType(FlatList);
  const items = tree.root.findAllByType(Button);

  expect(list).toHaveLength(1);
  expect(items).toHaveLength(9);
});

test('Click each item will move to separate screen', () => {
  const mockNavigation = { navigate: jest.fn() };
  const tree = renderer.create(<Home navigation={mockNavigation} />);
  const items = tree.root.findAllByType(Button);

  items[0].props.onPress();

  expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);

  items[1].props.onPress();

  expect(mockNavigation.navigate).toHaveBeenCalledTimes(2);

  items[2].props.onPress();

  expect(mockNavigation.navigate).toHaveBeenCalledTimes(3);

  items[3].props.onPress();

  expect(mockNavigation.navigate).toHaveBeenCalledTimes(4);
  // ...
});