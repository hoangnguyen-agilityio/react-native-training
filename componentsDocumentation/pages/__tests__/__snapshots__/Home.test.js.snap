// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "marginTop": 32,
      "paddingHorizontal": 24,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1",
          "page": "Indicator",
          "title": "Indicator",
        },
        Object {
          "id": "2",
          "page": "Buttons",
          "title": "Buttons",
        },
        Object {
          "id": "3",
          "page": "Lists",
          "title": "Lists",
        },
        Object {
          "id": "4",
          "page": "Views",
          "title": "Views",
        },
        Object {
          "id": "5",
          "page": "Modal",
          "title": "Modal",
        },
        Object {
          "id": "6",
          "page": "StatusBar",
          "title": "StatusBar",
        },
        Object {
          "id": "7",
          "page": "Switch",
          "title": "Switch",
        },
        Object {
          "id": "8",
          "page": "TextInput",
          "title": "TextInput",
        },
        Object {
          "id": "9",
          "page": "Touchable",
          "title": "Touchable",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Indicator
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Buttons
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Lists
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Views
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Modal
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                StatusBar
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Switch
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                TextInput
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Touchable
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
